stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -333,3 +398,68 @@ def _apply_strategy_full_bin(self, quants):
 
             if float_compare(need, location_quantity, rounding) != -1:
                 need = yield location, location_quantity, need, None, None
+
+    def _compare_with_tolerance(self, need, product_qty, rounding):
+        tolerance = self.tolerance_requested_value
+        limit = self.tolerance_requested_limit
+        computation = self.tolerance_requested_computation
+        if limit == "no_tolerance" or float_compare(tolerance, 0, rounding) == 0:
+            return float_compare(need, product_qty, rounding) == 0
+        elif limit == "upper_limit":
+            if computation == "percentage":
+                # need + rounding < product_qty <= need * (100 + tolerance) / 100
+                return (
+                    float_compare(need, product_qty, rounding) == -1
```

Comment:
Why you don't accept exact value when you give a tolerance? Should be in the limit defined by the tolerance but the quantity is exactly the need, it should also be valid, isn't it?
```suggestion
                    float_compare(need, product_qty, rounding) <= 0
```

Reply:
@elvise This still needs to be fixed or answered

Reply:
@francesco-ooops @geomer198 please take care

Reply:
> > @jbaudoux Done! Please check tests.
> 
> @geomer198 can you add tests with decimals (qty)?

Yes) Done!